vkubelet/env.go

```diff
@@ -341,6 +351,23 @@ loop:
 	return res, nil
 }
 
+// podFieldSelectorRuntimeValue returns the runtime value of the given
+// selector for a pod.
+func podFieldSelectorRuntimeValue(fs *corev1.ObjectFieldSelector, pod *corev1.Pod) (string, error) {
+	internalFieldPath, _, err := podshelper.ConvertDownwardAPIFieldLabel(fs.APIVersion, fs.FieldPath, "")
+	if err != nil {
+		return "", err
+	}
+	switch internalFieldPath {
+	case "spec.nodeName":
```

Comment:
Are these defined somewhere in the k8s codebase?

Reply:
I have followed the kubelet way of defining them. This can be referred for better understanding
https://github.com/kubernetes/kubernetes/blob/98fa2c7d32e59f798ff4201dfdcb439554392beb/pkg/kubelet/kubelet_pods.go#L785